File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
      int n = nums.size();
        long long current= 0;
        for(int i=0;i<k;i++){
            current+=(nums[i]);
        }
        int max=current;
        for(int i=k;i<n;i++){
            current+=nums[i]-nums[i-k];
            if (current>max){
            max=current;
            }
        }
        return (double)max/ k;
    }
    
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n=nums.length;
        double curr=0;
        for(int i=0;i<k;i++){
            curr+=nums[i];
        }
        double max=curr;
        for(int i=k;i<n;i++){
            curr+=nums[i]-nums[i-k];
            if(curr>max) max=curr;
            
        }
        return max/ k;
    }
}